kyo-prelude/shared/src/main/scala/kyo2/Abort.scala

```diff
@@ -83,23 +83,25 @@ object Abort:
                     )
                 }
             } {
-                case fail: E => Result.fail(fail)
-                case fail    => Result.panic(fail)
+                case fail: E if classOf[Throwable].isAssignableFrom(ct.runtimeClass) =>
+                    Result.fail(fail)
+                case fail => Result.panic(fail)
             }
     end RunOps
 
     inline def run[E >: Nothing]: RunOps[E] = RunOps(())
 
     final class CatchingOps[E <: Throwable](dummy: Unit) extends AnyVal:
-        def apply[A, B](v: => A < B)(
```

Comment:
Do we need `catching` if `run` also catches?

Reply:
`catching` is provides exception handling without handling the effect itself (return type is `A`, not `Result`). It's useful to isolate code that might throw exceptions and lift them into `Abort` suspensions.